docs/sphinx/source/whatsnew/v0.11.3.rst

```diff
@@ -31,6 +40,15 @@ Maintenance
 * asv 0.4.2 upgraded to asv 0.6.4 to fix CI failure due to pinned older conda.
   (:pull:`2352`)
 
+Breaking Changes
+~~~~~~~~~~~~~~~~
+* To ensure that the time zone in pvlib.location.Location remains internally
+  consistent if/when the time zone is updated, the tz attribute becomes the
+  single source of time-zone truth, is the single time-zone setter interface,
+  and its getter consistently returns an IANA string. datetime.tzinfo
+  subclasses (including time zones from pytz and zoneinfo) are still fully
+  supported in the tz setter, but the pytz attribute becomes read only.
+  (:issue:`2340`, :pull:`2341`)
```

Comment:
```suggestion
* The pytz attribute of pvlib.location.Location is now read only. Use the tz attribute
  to record time zone. The pytz attribute is internally updated if the tz attribute
  is changed. The tz attribute 
* The tz attribute of pvlib.location.Location now returns an IANA string.
  (:issue:`2340`, :pull:`2341`)
```
I think the focus of this note should be on how the changes will affect user code.

Reply:
I updated the whatsnew, trying to focus on the breaking change and its workaround, and also trying to provide the context behind the associated bugfix.

Reply:
Working on a second round of suggestions. The bug was that the docstring allowed for `tz` to be an int or float but inputs of those types weren't tested. The bug text in the last commit doesn't communicate that user code wasn't producing erroneous results.